auth/providers/azure/pop_tokenverifier_test.go

```diff
@@ -219,4 +231,17 @@ func TestPopTokenVerifier_Verify(t *testing.T) {
 		_, err := verifier.ValidatePopToken(invalidToken)
 		assert.Containsf(t, err.Error(), "Token is expired", "Error message is not as expected")
 	})
+
+	t.Run("'nonce' claim been reused - Cache validation", func(t *testing.T) {
+		nonceVerifier := azure.NewPoPVerifier("testHostname", 4*time.Second, -3*time.Second)
+		validToken, _ := azure.NewPoPTokenBuilder().SetTimestamp(time.Now().Unix()).SetHostName("testHostname").SetKid(azure.NonceClaimHardcoded).GetToken()
+		_, err := nonceVerifier.ValidatePopToken(validToken)
+		assert.NoError(t, err)
+		validToken, _ = azure.NewPoPTokenBuilder().SetTimestamp(time.Now().Unix()).SetHostName("testHostname").SetKid(azure.NonceClaimHardcoded).GetToken()
+		_, err = nonceVerifier.ValidatePopToken(validToken)
+		assert.Containsf(t, err.Error(), "Invalid token. 'nonce' claim is reused", "Error message is not as expected")
+		time.Sleep(2 * time.Second)
+		_, err = nonceVerifier.ValidatePopToken(validToken)
```

Comment:
nit: Add comments explaining each test (first usage is fresh nonce, second is reused so error, third is reused after expired so allowed).